runtime/perfetto/db.go

```diff
@@ -513,16 +515,28 @@ func (d *DB) Serve(ctx context.Context) error {
 	})
 	server := http.Server{Handler: mux}
 	ticker := time.NewTicker(time.Second)
+	var warnOnce sync.Once
 	for {
 		select {
 		case <-ticker.C:
 			lis, err := net.Listen("tcp", "localhost:9001")
-			if err == nil {
-				ticker.Stop()
-				return server.Serve(lis)
+			if err != nil {
+				warnOnce.Do(func() {
+					fmt.Fprintf(os.Stderr, "Perfetto server failed to listen on port 9001: %v\n", err)
+					fmt.Fprintf(os.Stderr, "Perfetto server will retry until port 9001 is available\n")
```

Comment:
I'm not sure if "Perfetto server" is the right term, or if this should be "Perfetto backend", or something else. I'm happy to adjust the wording here if anyone has suggestions.

Reply:
This sounds great to me.